src/prefect/cli/cloud.py

```diff
@@ -160,6 +160,10 @@ async def login(
                 for workspace in workspaces
             }
         except CloudUnauthorizedError:
+            if not key.startswith("pnu"):
+                exit_with_error(
+                    "Make sure you are using API key from https://app.prefect.cloud (Prefect 2) rather than https://cloud.prefect.io (Prefect 1)"
+                )
             exit_with_error(
                 "Unable to authenticate with Prefect Cloud. Please ensure your credentials are correct."
             )
```

Comment:
Something in this style would make more sense to me:

```suggestion
            help_message = "Please ensure your credentials are correct."
            if not key.startswith("pnu"):
                help_message = "Your key is not in our expected format."
            exit_with_error(
                "Unable to authenticate with Prefect Cloud. {help_message}"
            )
```

It seems nice retain the Cloud 1 messaging, but with your implementation if I provide the key "foo" I'll get a message about Prefect 1 vs 2 which doesn't make much sense. Is there a way we can detect that the key is from Cloud 1 from the format?

Reply:
> if I provide the key "foo" I'll get a message about Prefect 1 vs 2 which doesn't make much sense

It does make sense because the right Cloud2 key should start with `pnu_` and `foo` doesn't

Reply:
It doesn't make sense to hint that the key was retrieved from Cloud 1 when it's just arbitrary text. This could be confusing. I'd prefer to have separate warnings for these cases:
- Warn that it's not in the expected format if it doesn't start with the expected Cloud 2 prefix
- Warn that it's from the wrong application (1 vs 2) if it actually looks like a Cloud 1 key

Reply:
thx a lot, 100% agree. Does this look good or would you change the wording/structure?

![image](https://user-images.githubusercontent.com/86264395/192163320-e304640b-ce18-4dfe-8271-8af1a217474c.png)